core-test/src/test/scala/org/bitcoins/core/psbt/PSBTTest.scala

```diff
@@ -310,6 +310,22 @@ class PSBTTest extends BitcoinSAsyncTest {
     }
   }
 
+  it must "add Redeem Scripts and Witness Scripts to outputs" in {
+    forAllAsync(PSBTGenerators.fullNonFinalizedPSBT,
+                ScriptGenerators.randomNonP2SHScriptPubKey,
+                WitnessGenerators.scriptWitness) {
+      case (psbtF, spk, wit) =>
+        psbtF.map { psbt =>
+          val randOutput = scala.util.Random.nextInt(psbt.outputMaps.length)
```

Comment:
Should use `Gen.choose` instead of `Random.nextInt`

Reply:
AFAIK I cannot because I need the number of outputs when generating and I can't have a `Gen.choose` outside of the `forAllAsync`